### DISCUSSION TITLE
v3.2.0

### DISCUSSION THREAD

**Discussion by gnikit** [krvajal/vscode-fortran-support#566] [Announcements]
## What's Changed
* Bump lint-staged from 13.0.0 to 13.0.1 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/531
* Bump @typescript-eslint/parser from 5.27.1 to 5.28.0 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/528
* Bump webpack-cli from 4.9.2 to 4.10.0 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/533
* Bump @typescript-eslint/eslint-plugin from 5.27.1 to 5.28.0 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/532
* Bump @vscode/test-electron from 2.1.3 to 2.1.4 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/529
* Bump @types/node from 16.11.38 to 16.11.39 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/530
* Removes redundant tags & language aliases by @gnikit in https://github.com/fortran-lang/vscode-fortran-support/pull/537
* [BUG] `linter.modOutput` not working for Intel compilers by @gnikit in https://github.com/fortran-lang/vscode-fortran-support/pull/539
* Bump eslint from 8.17.0 to 8.18.0 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/541
* Bump lint-staged from 13.0.1 to 13.0.2 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/545
* Bump prettier from 2.6.2 to 2.7.1 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/543
* Bump eslint-plugin-jsdoc from 39.3.2 to 39.3.3 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/546
* Bump typescript from 4.7.3 to 4.7.4 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/544
* Bump @typescript-eslint/parser from 5.28.0 to 5.29.0 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/548
* Bump @typescript-eslint/eslint-plugin from 5.28.0 to 5.29.0 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/549
* Bump ts-loader from 9.3.0 to 9.3.1 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/551
* Bump @vscode/test-electron from 2.1.4 to 2.1.5 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/552
* Bump lint-staged from 13.0.2 to 13.0.3 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/558
* Bump eslint-plugin-prettier from 4.0.0 to 4.2.1 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/556
* Bump eslint from 8.18.0 to 8.19.0 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/557
* Bump @typescript-eslint/parser from 5.29.0 to 5.30.5 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/559
* Bump @typescript-eslint/eslint-plugin from 5.29.0 to 5.30.5 by @dependabot in https://github.com/fortran-lang/vscode-fortran-support/pull/560
* Add NAG linting by @gnikit in https://github.com/fortran-lang/vscode-fortran-support/pull/562
* Restructured test Fortran code by @gnikit in https://github.com/fortran-lang/vscode-fortran-support/pull/563
* Linter mod output by @gnikit in https://github.com/fortran-lang/vscode-fortran-support/pull/564
* chore: Refactors linter routines by @gnikit in https://github.com/fortran-lang/vscode-fortran-support/pull/565

**Full Changelog**: https://github.com/fortran-lang/vscode-fortran-support/compare/v3.1.0...v3.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/fortran-lang/vscode-fortran-support/releases/tag/v3.2.0'>v3.2.0</a>.</em>

**Comment by awvwgk** [comment]
Maybe worth to split the changelog in actual code base changes and dependency updates. The autogenerated release notes might be able to do this based on the labels on the PR ([docs](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#configuring-automatically-generated-release-notes)).

**Reply by gnikit** [reply]
I usually do the GitHub auto-generate release notes but for this release, because it contained so many changes and PRs from almost a year ago I chose not to, because I wanted only the changes that affect the users to be visible. Of course that is not 100% accurate and you can still see some maintenance changes in the official Changelog that users would not care, but I still think it is better than having a 100+ PRs from dependabot. The lesson learned from this was to never again take that long to issue a major release...

On a more general note, Javascript/Typescript projects have amazing dev tools, that autogenerate Changelogs and GitHub release notes, standardise commit messages, linting and lint fixing etc.. I spent a decent amount of time trying to find something that works but I never got around to finishing it.

Currently, we only have pre-commit hooks with husky for linting and formatting. It turns out the changelog/release notes and commit message standardisation is slightly harder to setup.
